pkgs/applications/version-management/gitkraken/default.nix

```diff
@@ -103,9 +106,11 @@ let
       comment = "Graphical Git client from Axosoft";
     }) ];
 
-    nativeBuildInputs = [ copyDesktopItems makeWrapper wrapGAppsHook ];
+    nativeBuildInputs = [ copyDesktopItems makeWrapper wrapGAppsHook cacert ];
     buildInputs = [ gtk3 gnome.adwaita-icon-theme ];
 
+    env.GIT_SSL_CAINFO = "${cacert}/etc/ssl/certs/ca-bundle.crt";
```

Comment:
sorry, i just realised, this probably doesn't actually work,
these are set at build time, but i assume we want to set it at runtime

Reply:
I tested it, it works with or without `env.` in front. What should I do then?

Reply:
Hmm Those are only set at built time. So either something captures them during build or test is not right.  Is there a public repository you can reproduce issue with ?

Reply:
Well I’m testing it by building from [my nixpkgs repository branch `gitkraken-bundled`](https://github.com/nicolas-goudry/nixpkgs/tree/gitkraken-bundled-git) with the following command:
```
NIXPKGS_ALLOW_UNFREE=1 nix-build -A gitkraken
```

Then I pick up the directory and run:
```
/nix-store/<build-dir>/bin/gitkraken
```

I made sure to force the rebuild by adding/updating an `echo` inside the installPhase.

Reply:
Huh… Something is not right. I removed the `GIT_SSL_CAINFO` env var altogether and it’s still working.

Reply:
Ok, I managed to reproduce the SSL issue inside a NixOS VM. I don’t understand why it’s not reproducible on my host, but at least I can work from there.

FYI, neither `env.GIT_SSL_CAINFO` nor plain `GIT_SSL_CAINFO` did work. If I run Gitkraken while setting the `GIT_SSL_CAINFO` env var, it works as expected.

Reply:
I tried several things:
- `cacert` as `nativeBuildInputs`
- `cacert` as `buindInputs`
- `cacert` neither in `nativeBuildInputs` nor `buildInputs`
- setting `GIT_SSL_CA_INFO = "${cacert}/..."`
- setting `env.GIT_SSL_CA_INFO = "${cacert}/..."`
- using `makeWrapperArgs+=("--prefix" "GIT_SSL_CAINFO" ":" "${cacert}/...")` in `postInstall`

None of them worked. In the end, the only way I managed to make it work was a variant of what I did previously:
```
pushd $out/share/${pname}/resources/app.asar.unpacked/git
for dir in bin libexec share; do
  rm -rf $dir
  ln -s ${git}/$dir $dir
done
rm -rf ssl/cacert.pem
ln -s ${cacert}/etc/ssl/certs/ca-bundle.crt ssl/cacert.pem
popd
```

I know it’s overkill, but I can’t find another way to make it work… If you have any suggestions, I’m all ears.

Reply:
makeWrapper is the only way to go, as you have pointed out

>  Gitkraken while setting the GIT_SSL_CAINFO env var, it works as expected

can you provide a VM or a clone repo url to try to reproduce the issue ?

Reply:
Not sure if that’s what you asked, but here is a repository containing the self-contained GitKraken package with instructions to build it and reproduce the issue: https://github.com/nicolas-goudry/nix-gitkraken-with-bundled-git-ssl-issue

Reply:
I am interested reproducing 

> Push gitkraken-bundled-git: fatal: unable to access 'https://github.com/nicolas-goudry/nixpkgs.git/': error setting certificate file: /nix/store/l02yd3ssx9qyfljxf0m5yzyw7gyq1g88-gitkraken-9.10.0/share/gitkraken/resources/app.asar.unpacked/git/ssl/cacert.pem

What I mean by repository I mean repository to reproduce push / clone issue. I  dont need a repo to build gitkraken

Reply:
Oh ok, just created a [`blank` repository](https://github.com/nicolas-goudry/blank) and invited you as a collaborator.

Reply:
I thought It could be reproduced with public repository, either way. Cloning via gitkraken worked for me, without the need to set ssl CA cert.

So, unless we can reproduce the issue, there is no need to do anything

Reply:
The issue appears when pushing to the repository, so you need write access to do that (hence the invite)

Reply:
this seems to work

```
@@ -118,7 +166,6 @@ let
       cp -R $src/* $out/share/${pname}
 
       mkdir -p $out/bin
-      ln -s $out/share/${pname}/${pname} $out/bin/
 
       mkdir -p $out/share/pixmaps
       cp ${pname}.png $out/share/pixmaps/${pname}.png
@@ -127,6 +174,8 @@ let
     '';
 
     postFixup = ''
+      makeWrapper $out/share/${pname}/${pname} $out/bin/gitkraken --set GIT_SSL_CAINFO "${cacert}/etc/ssl/certs/ca-bundle.crt"
+
```

Reply:
Indeed it works! I’ll update the PR with this.

Should I include the `cacert` package in the `buildInputs` ?

Reply:
no need to add `cacert` into `buildInputs` as its referenced explicitly.

othewise, let me know if there are any other questions/issues

Reply:
Ok then I won’t touch this PR further as everything seems to be working.

I’ll install it as my daily driver to see if there are any other issues, but I believe there won’t be.

As a side note, I’ll wait for the [update PR](https://github.com/NixOS/nixpkgs/pull/272139) to be merged in order to remove the duplicate commits.

Thanks a lot for your help on this.